docs/attributes-registry/log.md

```diff
@@ -40,12 +40,71 @@ Attributes for a file to which log was emitted.
 
 This document defines the generic attributes that may be used in any Log Record.
 
-| Attribute             | Type   | Description                                 | Examples                                                                                                                         | Stability                                                        |
-| --------------------- | ------ | ------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------- | ---------------------------------------------------------------- |
-| `log.record.original` | string | The complete orignal Log Record. [1]        | `77 <86>1 2015-08-06T21:58:59.694Z 192.168.2.133 inactive - - - Something happened`; `[INFO] 8/3/24 12:34:56 Something happened` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
-| `log.record.uid`      | string | A unique identifier for the Log Record. [2] | `01ARZ3NDEKTSV4RRFFQ69G5FAV`                                                                                                     | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
-
-**[1]:** This value MAY be added when processing a Log Record which was originally transmitted as a string or equivalent data type AND the Body field of the Log Record does not contain the same value. (e.g. a syslog or a log record read from a file.)
-
-**[2]:** If an id is provided, other log records with the same id will be considered duplicates and can be removed safely. This means, that two distinguishable log records MUST have different values.
-The id MAY be an [Universally Unique Lexicographically Sortable Identifier (ULID)](https://github.com/ulid/spec), but other identifiers (e.g. UUID) may be used as needed.
+| Attribute             | Type   | Description                                  | Examples                                                                                                                         | Stability                                                        |
+| --------------------- | ------ | -------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------- | ---------------------------------------------------------------- |
+| `log.record.id`       | int    | A durable identifier for the Log Record. [1] | `1`; `10018`                                                                                                                     | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| `log.record.name`     | string | A durable name for the Log Record. [2]       | `RequestProcessed`; `InvalidResponse`                                                                                            | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
```

Comment:
How are `id` and `name` related? Is `name` supposed to be the human-readable equivalent of `id`? Can different `id`-ed logrecords have the same `name`?

Reply:
> Is name supposed to be the human-readable equivalent of id

Yes.

> Can different id-ed logrecords have the same name?

Not in well behaved application.

Reply:
Based on today discussion in the Event SIG, I believe we should use `event.name` for the durable event name. 

The TL;DR of the discussion

If logging API allows to define event name (like .NET, C++ and maybe Rust do), e.g.

```csharp
logger.LogInformation(new EventId(id: 42, name: "com.foo.my-event-name"), "something important")
```

then the corresponding name should be recorded as an `event.name`.

It's a user responsibility to have a unique name, there is nothing we can do to enforce it if users call Event API directly. 

So if user does `logger.LogInformation(new EventId(42, name: "foo"), "something important")` the best that logging bridge can do is to use the name that user provided and set `event.name = foo`. 

I wonder if other participants share the same understanding @CodeBlanch @MSNev @trask @breedx-splk

Reply:
Related to this, [early versions of Log data model](https://github.com/open-telemetry/oteps/blob/main/text/logs/0097-log-data-model.md#field-name) had a `Name` field defined as:

>Short event identifier that does not contain varying parts. Name describes what happened (e.g. "ProcessStarted"). Recommended to be no longer than 50 characters. Not guaranteed to be unique in any way. Typically used for filtering and grouping purposes in backends. This field is optional.

This seems pretty close to what the proposed `log.record.name` is and also to existing `event.name`. It almost feels like we should have help the `Name` field on LogRecord. I [commented here](https://github.com/open-telemetry/opentelemetry-specification/issues/3406#issuecomment-2307268466) that I think we need consider adding it back and then it would replace `event.name` and the proposed `log.record.name`.

Reply:
> Based on today's discussion in the Event SIG, I believe we should use `event.name` for the durable log name.
> 
> The gist of the discussion:
> 
> If logging API allows to define event name (like .NET, C++ and maybe Rust do), e.g.
> 
> ```cs
> logger.LogInformation(new EventId(id: 42, name: "com.foo.my-event-name"), "something important")
> ```
> 
> then the corresponding name should be recorded as an `event.name`.
> 
> It's a user responsibility to provide a unique name, there is nothing we can do to enforce it if users call Event API directly.
> 
> So if user does `logger.LogInformation(new EventId(42, "foo"), "something important")` the best that logging bridge can do is to use the name that user provided and set `event.name = foo`.
> 
> I wonder if other participants share the same understanding @CodeBlanch @MSNev @trask @breedx-splk

This is effectively what we talked about, however, as mentioned in the call my point of view (and recommendation) is still that otel "should" define that for this scenario there is a generic prefix that is prepended to `event.name` to ensure that these types of `events` cannot directly clash with any other OTel defined event.

During the meeting I proposed `log.*` but perhaps better options would be either `app.*`, `user.*` or `custom.*` so that it is explicitly called for backend processors of "known" event names that this is a custom event and therefore should be treated as such, as opposed to requiring them to enforce (more explicit) validity checks on any possible "known" events than necessary).

This should also NOT preclude the ability of languages reusing their existing logging mechanism's to "generate" true valid OTel defined events (without any prefix automatically added). Which could be done (potentially) by the introduction of some other explicitly named "element" (class, provider, property, attribute etc)

Reply:
is the goal of the prefix to prevent users from accidentally using a name that's defined in otel?

We should give user the same guidance that we give for attribute names - 

https://github.com/open-telemetry/semantic-conventions/blob/3c16c802e8ae8849ae0cf31eac02c3cabf64e4dd/docs/general/attribute-naming.md?plain=1#L132-L136

We don't attempt to do anything else for attributes, why we would do it for events?

Reply:
I think that changing or eliminating `event.name` would be a step backwards at this point. 

I don't think that the existence of 3rd party event APIs that have two identifiers for an event (one numeric, one readable) should influence the design we've already arrived at in otel. Just put the readable id into the `event.name` attribute and put the numeric id into a long attribute named whatever non-semconv thing you want. Why does it need to be more complicated than that....or are we trying to provide formal mapping semantics for every other Event system that might exist out there?

Reply:
> It almost feels like we should have kept the Name field on LogRecord. I https://github.com/open-telemetry/opentelemetry-specification/issues/3406#issuecomment-2307268466 that I think we need consider adding it back and then it would replace event.name and the proposed log.record.name.

+1 . Irrespective of that, in implementations,  (.NET, Rust), we store name as a top-level field in `LogRecord`, and while exporting, its sent via attributes.

Reply:
> I think that changing or eliminating event.name would be a step backwards at this point.

That is not part of the proposal, right?


> Just put the readable id into the event.name attribute and put the numeric id into a long attribute named whatever non-semconv thing you want.

This is not just a .NET specific thing, else we could have just used attribute named "dotnet.ilogger.eventid.id"/similar. The goal is to get something language neutral. .NET's ILogger was used as an example.

OTel C++ has the exact same need.
OTel Rust don't yet have the numerical identifier (it just has the human readable name part), but we are trying to get it added as well.

Reply:
I agree, it seems we have 3 different discussions here
1. use `event.name` vs introduce new `log.record.name`
2. do we need first-class data model Name property on logs
3. how to capture id (that appears in several languages and also in windows event log).

Maybe we can focus on the p1 in this thread

Reply:
Yes back on topic! `log.record.name`!

My suggestion is that we separate `log.record.name` and `log.record.id`. The ID attribute is clear cut and valuable, not to mention a big deal. My request would be to please reduce this PR so that it only contains the ID attribute.  That can go ahead and be approved based on its own merits.

If `log.record.name` really is incompatible with our definition of `event.name`, please make a separate PR for `log.record.name` that clearly presents which issues the additional name field would address. ✌️ 

### on the name field thing
Philosophically I agree that Name should be a field on LogRecord. That would match the way we handle it on Span. But in this particular case, I believe that the Name field actually complicates things. 

This is the problem: we want to support the use case where 3rd party loggers can create events on their own. Those loggers have no concept of a `Name` field, so they would have to use the `event.name` attribute pattern.  Which means that there would have to be code in various places that scans for the `event.name` attribute, and then tries to delete that attribute and move the value over to the Name field.

Because support for 3rd party loggers is critical, there is no escaping the existence of the `event.name` attribute in many parts of our logging pipeline. Which means that the choice is not between "do you want to only have an `event.name` attribute?" and "do you want to only have an `Name` field?", it is actually between "do you want to only have an `event.name` attribute?" or "do you want to have both an `event.name` attribute plus a `Name` field you have to check?"

Long story short: It appears to be the case that a lot of double checking and confusion could be avoided if we just stick with the attribute `event.name`. Solving problems by doing nothing, this is The Way. :)

Reply:
Based on these discussions/feedback, this is my understanding:
1. There is general agreement that there is no need of introducing `log.record.name` as it is same as the `event.name`. That makes every log done using .NET's ILogger with EventName (_eg: logger.Log(new EventId(10, "LoginFailed"), body, attributes);_) an OTel Event, as it is a `LogRecord` with an attribute "event.name". No need of spec/semantic convention changes, OTel .NET can fix its implementation to store EventName as "event.name".

2. The numerical version of the "event.name", could be named "event.id", call out that "event.id" and "event.name" should have 1:1 relation in a well behaved app. This can be a separate PR.

example:
```
var logger = GetLogger("company.product.service.module");
logger.Log(new EventId(10, "LoginFailed"), body, attributes);
```

The above will produce a LogRecord with scope name set to "company.product.service.module", and attribute containing "event.name" = "LoginFailed", and "event.id" = 10 (along with body and other attributes)

Is this everyone's understanding as well?